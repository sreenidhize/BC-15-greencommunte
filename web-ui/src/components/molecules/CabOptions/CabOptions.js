import React from 'react';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from "@material-ui/core";
import PropTypes from "prop-types";

const useStyles = makeStyles((theme) => ({
    primary: {
        fontSize: "14px",
        color: "#19293b",
    },
    secondary: {
        fontSize: "14px",
        color: "#5f7381",
    },
    theme: {
        fontSize: "14px",
        color: theme.palette.primary.main,
        fontWeight: "600",
    },
    img: {
        marginRight: "10px",
        height: "42px",
        width: "42px",
    },
    gird: {
        minWidth: "800px"
    }
}));

const CabOptions = ({options}) => {
    const classes = useStyles();

    const cabs = options.map((obj) => {
        return (
            <Grid container direction="row" alignItems="center" key={obj.type}>
                    <img className={classes.img} src={obj.url} alt={obj.type}></img>
                <Grid item xs={3} container direction="column" justify="center">
                    <Typography variant="h4" className={classes.primary}>{obj.type}</Typography>
                    <Typography variant="h4" className={classes.secondary}>Approx {obj.value}</Typography>
                </Grid>
                    <Typography variant="h4" className={classes.theme}>Book now</Typography>
            </Grid>
        );
    });

    return (
        <Grid container direction="column" justify="center" data-testid="cabOptions-placeholder">
            {cabs}
        </Grid>
    );
};

CabOptions.propTypes = {
    options: PropTypes.array,

};

export default CabOptions;