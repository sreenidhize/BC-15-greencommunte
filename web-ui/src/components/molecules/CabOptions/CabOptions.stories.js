import React from 'react';
import PropTypes from 'prop-types';
import CabOptions from './CabOptions';
import theme from '../../../utils/theme/theme';
import { ThemeProvider } from '@material-ui/core';

export default {
    title: 'molecules/CabOptions',
    component: CabOptions,
};

const Template = (args) => <ThemeProvider theme={theme} ><CabOptions {...args} /></ThemeProvider>;

export const Options = Template.bind({});
Options.args = {
    options: [
        { type: "OLA", value: "40", url: "https://res-5.cloudinary.com/crunchbase-production/image/upload/c_lpad,h_256,w_256,f_auto,q_auto:eco/v1459446250/xx0u5xwwzygivm67ogsk.jpg" },
        { type: "UBER", value: "40", url: "https://semaine-production.s3.amazonaws.com/products/product/image/6827/large_Uber-app.jpg" },
        { type: "Rapido", value: "40", url: "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSWOLJJfUAbhh1JVtRLgNCexBH9_cC2L3NvqA&usqp=CAU" }]
};
