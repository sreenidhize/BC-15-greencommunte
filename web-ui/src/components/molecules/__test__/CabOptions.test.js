import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import CabOptions from '../CabOptions/CabOptions';

const options= [
  { type: "OLA", value: "40", url: "https://res-5.cloudinary.com/crunchbase-production/image/upload/c_lpad,h_256,w_256,f_auto,q_auto:eco/v1459446250/xx0u5xwwzygivm67ogsk.jpg" },
  { type: "UBER", value: "40", url: "https://semaine-production.s3.amazonaws.com/products/product/image/6827/large_Uber-app.jpg" },
  { type: "Rapido", value: "40", url: "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSWOLJJfUAbhh1JVtRLgNCexBH9_cC2L3NvqA&usqp=CAU" }
];

describe("CabOptions placeholder test", () => {
  it("Matches the snapshot", () => {
    const { container } = render(
      <CabOptions options={options} />
    );
    expect(container).toMatchSnapshot();
  });

  test("test CabOptions render", () => {
    const { getByTestId } = render(<CabOptions options={options} />);
    expect(getByTestId("cabOptions-placeholder")).toBeInTheDocument();
  });
});