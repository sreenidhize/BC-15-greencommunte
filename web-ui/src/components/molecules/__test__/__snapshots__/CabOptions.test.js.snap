// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CabOptions placeholder test Matches the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-justify-xs-center"
    data-testid="cabOptions-placeholder"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <img
        alt="OLA"
        class="makeStyles-img-4"
        src="https://res-5.cloudinary.com/crunchbase-production/image/upload/c_lpad,h_256,w_256,f_auto,q_auto:eco/v1459446250/xx0u5xwwzygivm67ogsk.jpg"
      />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-justify-xs-center MuiGrid-grid-xs-3"
      >
        <h4
          class="MuiTypography-root makeStyles-primary-1 MuiTypography-h4"
        >
          OLA
        </h4>
        <h4
          class="MuiTypography-root makeStyles-secondary-2 MuiTypography-h4"
        >
          Approx 
          40
        </h4>
      </div>
      <h4
        class="MuiTypography-root makeStyles-theme-3 MuiTypography-h4"
      >
        Book now
      </h4>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <img
        alt="UBER"
        class="makeStyles-img-4"
        src="https://semaine-production.s3.amazonaws.com/products/product/image/6827/large_Uber-app.jpg"
      />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-justify-xs-center MuiGrid-grid-xs-3"
      >
        <h4
          class="MuiTypography-root makeStyles-primary-1 MuiTypography-h4"
        >
          UBER
        </h4>
        <h4
          class="MuiTypography-root makeStyles-secondary-2 MuiTypography-h4"
        >
          Approx 
          40
        </h4>
      </div>
      <h4
        class="MuiTypography-root makeStyles-theme-3 MuiTypography-h4"
      >
        Book now
      </h4>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <img
        alt="Rapido"
        class="makeStyles-img-4"
        src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSWOLJJfUAbhh1JVtRLgNCexBH9_cC2L3NvqA&usqp=CAU"
      />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-justify-xs-center MuiGrid-grid-xs-3"
      >
        <h4
          class="MuiTypography-root makeStyles-primary-1 MuiTypography-h4"
        >
          Rapido
        </h4>
        <h4
          class="MuiTypography-root makeStyles-secondary-2 MuiTypography-h4"
        >
          Approx 
          40
        </h4>
      </div>
      <h4
        class="MuiTypography-root makeStyles-theme-3 MuiTypography-h4"
      >
        Book now
      </h4>
    </div>
  </div>
</div>
`;
